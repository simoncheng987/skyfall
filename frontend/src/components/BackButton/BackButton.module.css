.backButton {
  position: absolute;
  width: 50px;
  top: 60px;
  left: 60px;
  transition: all 0.2s ease-in-out;
  background-color: transparent;
  border: none;
  cursor: pointer;
}

.backButton:hover {
  opacity: 0.5;
}
