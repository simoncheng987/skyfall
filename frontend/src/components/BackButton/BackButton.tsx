import { ArrowSmLeftIcon } from '@heroicons/react/solid';
import React from 'react';
import { useNavigate } from 'react-router-dom';
import styles from './BackButton.module.css';

export default function BackButton() {
  const navigate = useNavigate();
  return <ArrowSmLeftIcon className={styles.backButton} onClick={() => navigate('/')} />;
}
