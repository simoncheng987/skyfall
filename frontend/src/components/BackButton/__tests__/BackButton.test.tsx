import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter, Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { createMemoryHistory, MemoryHistory } from 'history';
import BackButton from '../BackButton';

function backButtonWithRouterFactory(memoryHistory: MemoryHistory) {
  return (
    <Router location={memoryHistory.location} navigator={memoryHistory}>
      <BackButton />
    </Router>
  );
}

function createComponentWithRouter() {
  const memoryHistory = createMemoryHistory({ initialEntries: ['/', '/test-entry'] });
  render(backButtonWithRouterFactory(memoryHistory));
  return memoryHistory;
}

describe('BackButton', () => {
  test('the back button is rendered on the screen', () => {
    createComponentWithRouter();
    expect(screen.getByLabelText('back-button')).toBeInTheDocument();
  });

  test('snapshot test of the rendered back button', () => {
    const tree = renderer
      .create(
        <MemoryRouter initialEntries={[{ pathname: '/test-pathname', key: 'test-key' }]}>
          <BackButton />
        </MemoryRouter>,
      )
      .toTree();
    expect(tree).toMatchSnapshot();
  });

  test('clicking the back button redirects user to root path (/)', () => {
    const mockHistory = createComponentWithRouter();
    const backButton = screen.getByLabelText('back-button');
    fireEvent.click(backButton);
    expect(mockHistory.location.pathname).toEqual('/');
  });
});
