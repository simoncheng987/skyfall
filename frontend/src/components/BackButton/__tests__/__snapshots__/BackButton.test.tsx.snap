// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton snapshot test of the rendered back button 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "children": <BackButton />,
    "initialEntries": Array [
      Object {
        "key": "test-key",
        "pathname": "/test-pathname",
      },
    ],
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "component",
    "props": Object {
      "basename": undefined,
      "children": <BackButton />,
      "location": Object {
        "hash": "",
        "key": "test-key",
        "pathname": "/test-pathname",
        "search": "",
        "state": null,
      },
      "navigationType": "POP",
      "navigator": Object {
        "action": "POP",
        "back": [Function],
        "block": [Function],
        "createHref": [Function],
        "forward": [Function],
        "go": [Function],
        "index": 0,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "test-key",
          "pathname": "/test-pathname",
          "search": "",
          "state": null,
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "rendered": Object {
      "instance": null,
      "nodeType": "component",
      "props": Object {},
      "rendered": Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "aria-label": "back-button",
          "children": <ForwardRef(ArrowSmLeftIcon) />,
          "className": "backButton",
          "onClick": [Function],
          "type": "button",
        },
        "rendered": Array [
          Object {
            "instance": null,
            "nodeType": "host",
            "props": Object {
              "aria-hidden": "true",
              "children": <path
                clipRule="evenodd"
                d="M9.707 14.707a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 1.414L7.414 9H15a1 1 0 110 2H7.414l2.293 2.293a1 1 0 010 1.414z"
                fillRule="evenodd"
              />,
              "fill": "currentColor",
              "viewBox": "0 0 20 20",
              "xmlns": "http://www.w3.org/2000/svg",
            },
            "rendered": Array [
              Object {
                "instance": null,
                "nodeType": "host",
                "props": Object {
                  "clipRule": "evenodd",
                  "d": "M9.707 14.707a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 1.414L7.414 9H15a1 1 0 110 2H7.414l2.293 2.293a1 1 0 010 1.414z",
                  "fillRule": "evenodd",
                },
                "rendered": Array [],
                "type": "path",
              },
            ],
            "type": "svg",
          },
        ],
        "type": "button",
      },
      "type": [Function],
    },
    "type": [Function],
  },
  "type": [Function],
}
`;
